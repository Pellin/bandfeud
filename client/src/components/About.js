import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { motion } from 'framer-motion';

import BFLogoSvg from '../icons/BFLogoSvg';
import LogoSquareSvg from '../icons/LogoSquareSvg';

const About = ({ os }) => {
  return (
    <motion.div
      className="info-container"
      initial={{ opacity: 0 }}
      animate={{ opacity: 1 }}
      transition={{ duration: 1 }}
    >
      {os === 'desktop' ? (
        <BFLogoSvg className="background-logo" />
      ) : (
        <LogoSquareSvg className="background-logo-square" />
      )}
      <div className="info">
        <h1>How to play</h1>

        <h2>Your turn</h2>
        <p>
          Type in a band or an artist beginning with the letter visible under
          the input field.
        </p>
        <p>
          Band or artist names must contain at least two letters or numbers. It
          must begin and end with a letter between a and z or a number.
        </p>
        <p>
          Wait while the app checks if the name exists in the Discogs database.
        </p>
        <p>
          If the name exists, your score increases. The longer the band name,
          the more points you get.
        </p>
        <h2>App turn</h2>
        <p>Next, the app will present a band or an artist. </p>
        <p>
          You now need to submit a band or artist name that begins with the last
          letter of the name the app presented.
        </p>
        <h2>Detailed instructions</h2>
        <p>
          For more detailed instructions and references, please visit the{' '}
          <a
            rel="noopener noreferrer"
            target="_blank"
            href="https://github.com/Pellin/bandfeud"
          >
            Bandfeud Github page
          </a>
          .
        </p>
      </div>
      <Link to="/" className="back-button">
        BACK
      </Link>
    </motion.div>
  );
};

const mapStateToProps = state => ({
  os: state.os
});

export default connect(mapStateToProps)(About);
