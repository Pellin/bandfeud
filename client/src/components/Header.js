import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { motion } from 'framer-motion';

import Counter from '../components/home/play/Counter';
import Score from '../components/home/play/Score';
import BFLogoSvg from '../icons/BFLogoSvg';
import ISvg from '../icons/ISvg';
import CrownSvg from '../icons/CrownSvg';

export const Header = ({
  onAddFailedBand,
  buttonPressed,
  inGame,
  submitted,
  used,
  difficulty,
  message,
  score,
  os
}) => {
  const visibleVariants = {
    show: {
      opacity: 1
    },
    hide: {
      opacity: 0
    }
  };
  return (
    <div className="header-container">
      <div className="header-side">
        {!inGame && (
          <motion.div
            className="i-link"
            initial={{ opacity: 0 }}
            animate={buttonPressed ? 'hide' : 'show'}
            transition={{ duration: 1.5 }}
            variants={visibleVariants}
          >
            <Link to="/about/">
              <ISvg />
            </Link>
          </motion.div>
        )}
        {inGame && !submitted && !message && os === 'desktop' && (
          <div className="counter-container">
            <Counter
              addFailedBand={onAddFailedBand}
              submitted={submitted}
              inGame={inGame}
              used={used}
              difficulty={difficulty}
              score={score}
            />
          </div>
        )}
      </div>
      <div className="header-center">
        {inGame && (
          <div className="header-logo">
            <BFLogoSvg />
          </div>
        )}
      </div>
      <div className="header-side">
        {!inGame && (
          <motion.div
            className="crown-link"
            initial={{ opacity: 0 }}
            animate={buttonPressed ? 'hide' : 'show'}
            transition={{ duration: 1.5 }}
            variants={visibleVariants}
          >
            <Link to="/highscores/">
              <CrownSvg className="crown-right" />
            </Link>
          </motion.div>
        )}
        {inGame && os === 'desktop' && (
          <div className="desktop-score">
            <Score score={score} />
          </div>
        )}
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  inGame: state.inGame,
  submitted: state.submitted,
  used: state.used,
  difficulty: state.difficulty,
  message: state.message,
  score: state.score,
  os: state.os
});

const mapDispatchToProps = dispatch => ({
  onAddFailedBand: band =>
   dispatch({
     type: 'ADD_FAILED_BAND',
     payload: { name: band, mode: 'Out of time' }
   }),
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
