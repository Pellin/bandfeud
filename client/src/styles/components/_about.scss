.info-container {
  background: url('../../icons/BFLogoSvg.js');
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 90vh;
  position: fixed;
  width: 100vw;

  @media (min-width: $desktop-breakpoint) {
    height: 100vh;
    overflow: hidden;
  }

  @media (max-height: 675px) and (min-width: 570px) {
    height: 100vh;
    overflow: hidden;
  }
}

.info {
  color: lighten($off-white, 10%);
  height: 60vh;
  width: 85vw;
  overflow: hidden;

  a {
    color: white;
    text-decoration: none;
  }

  h1,
  h2 {
    color: $off-white;
    margin-bottom: 0;
  }

  p {
    margin: 0.5rem 0 1rem;
    line-height: 1.4;
  }

  @media (min-width: $desktop-breakpoint) {
    font-size: 1.3rem;
    height: 55rem;
    width: 32rem;
  }

  @media (max-height: 675px) and (min-width: 570px) {
    font-size: 1.2rem;
    width: 36rem;
  }
}

.about-logo {
  height: auto;
  width: 35vw;
  position: relative;
  top: 2vh;
  margin-left: 1vh;

  @media(min-width: $desktop-breakpoint) {
    width: 14vw;
    margin-left: 0.5vw;
    top: 2.5vh;
  }
}