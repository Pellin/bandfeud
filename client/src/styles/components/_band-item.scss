.band-item {
  align-items: center;
  display: flex;
  max-height: 20rem;
  flex-direction: column;
  justify-content: center;
  margin: 0.5rem;
  width: 20rem;
}

.image-container {
  align-items: center;
  display: flex;
  height: 23rem;
  justify-content: center;
  margin: 0 0 3.5rem;

  @media (min-width: $desktop-breakpoint) {
    height: 22rem;
    margin: 0 0 1rem;
  }

  @media (max-width: 568px) {
    height: 21rem;
  }
}

.band-item-image {
  border-radius: 5px;
  width: 18rem;

  @media (min-width: $desktop-breakpoint) {
    width: 20rem;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    width: 17rem;
  }

  @media (max-width: 568px) {
    width: 16rem;
  }

  //iphone 11 pro & iphone X & iphone Xs
  @media only screen and (min-device-width: 375px) and (min-device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait) {
    width: 19rem;
  }
  //iphone 11 & iphone XR
  @media only screen and (device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) {
    width: 19rem;
  }
}

.img {
  width: 100%;
  max-height: 23rem;
  border-radius: 5px;

  @media (min-width: $desktop-breakpoint) {
    max-height: 30rem;
  }
}

.title-flow {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 3rem;
  justify-content: center;
  margin-top: 20rem;
  position: absolute;

  @media (min-width: $desktop-breakpoint) {
    margin-top: $desktop-top;
  }

  @media (max-width: 568px) {
    margin-top: 12rem;
  }
}

.title-flow-firefox {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 3rem;
  justify-content: center;
  position: absolute;
  margin-top: 32rem;
}

.band-item-title-container {
  width: 20rem;
  margin-top: 3.5rem;
  text-align: center;

  @media (min-width: $desktop-breakpoint) {
    margin-top: 2rem;
  }
}

.band-item-title {
  font-family: 'Righteous', cursive;
  font-size: 1.8rem;
  text-transform: uppercase;

  @media (min-width: $desktop-breakpoint) {
    font-size: 2.5rem;
  }

  //iphone 11 pro & iphone X & iphone Xs
  @media only screen and (min-device-width: 375px) and (min-device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait) {
    font-size: 2rem;
  }
  //iphone 11 & iphone XR
  @media only screen and (device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) {
    font-size: 2rem;
  }
}

.points {
  color: $yellow;
  font-size: 2.5rem;
  opacity: 1;
  position: absolute;
  top: 6rem;
  transition: opacity, 1.5s;
}

.points-hide {
  opacity: 0;
  position: absolute;
  top: 6rem;
  transition: opacity, 1.5s;
}
