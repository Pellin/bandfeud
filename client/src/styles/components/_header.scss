.header-container {
  border-bottom: 1px solid $off-white;
  position: fixed;
  top: 0px;
  display: flex;
  justify-content: space-evenly;
  width: 100vw;
  z-index: 100;
}

.flex-counter {
  width: 33%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.counter-container {
  background: transparent;
  text-align: center;
  height: 4rem;
  width: 4rem;
}

.counter {
  align-items: center;
  color: $yellow;
  display: flex;
  font-size: 2rem;
  width: 2rem;
  height: 3rem;

  @media (min-width: $desktop-breakpoint) {
    font-size: 3rem;
    width: 3rem;
    height: 4rem;
  }
}

.header-center {
  align-items: center;
  flex-direction: row;
  display: flex;
  justify-content: space-around;
  margin: auto;
  padding: 0.5rem;
  width: 80%;
  z-index: 10;
}

.header-logo {
  cursor: default;
  margin: 5px;
  width: 100px;

  img {
    width: 100%;
    height: auto;
  }
  @media (min-width: $desktop-breakpoint) {
    margin: 10px;
    width: 200px;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    width: 150px;
  }
}

.header-logo-link {
  cursor: pointer;
  margin: 5px;
  width: 100px;

  img {
    width: 100%;
    height: auto;
  }
  @media (min-width: $desktop-breakpoint) {
    margin: 10px;
    width: 200px;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    width: 150px;
  }
}

.link-container {
  display: flex;
  align-items: center;
  margin: 0;
  padding: 0;
  text-align: center;
  justify-content: center;
  width: 20%;
}

.link {
  font-size: 0.9rem;
  margin-left: 0.8rem;

  a {
    color: $yellow;
    text-decoration: none;
  }

  @media (min-width: $desktop-breakpoint) {
    font-size: 1.2rem;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    font-size: 1rem;
  }
}

.active-link {
  font-size: 1.1rem;
  color: #efeefe;
  text-align: center;

  @media (min-width: $desktop-breakpoint) {
    font-size: 1.4rem;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    font-size: 1.2rem;
  }
}

.flex-score {
  width: 33%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
}

.score {
  color: $orange;
  font-size: 1.5rem;

  @media (min-width: $desktop-breakpoint) {
    font-size: 2.3rem;
  }
}
