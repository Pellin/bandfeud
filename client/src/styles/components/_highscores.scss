.highscore-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: auto;
  position: fixed;
  top: 8rem;
  left: 0;
  right: 0;

  @media (min-width: $desktop-breakpoint) {
    top: $desktop-top + 2rem;
  }

  @media (min-height: 820px) {
    top: 23rem;
  }

  @media (max-height: 740px) and (min-width: 570px) {
    top: $desktop-top - 3rem;
  }
}

.highscore-title {
  color: $orange;
  cursor: default;
  font-size: 2.3rem;
  font-weight: 900;
  letter-spacing: 9px;
  margin-bottom: 1.5rem;
  padding-left: 9px;
  text-align: center;

  @media (max-height: 568px) {
    font-size: 1.7rem;
    margin-bottom: 1rem;
    letter-spacing: 7px;
    padding-left: 7px;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    font-size: 2rem;
  }

  @media (min-height: 820px) {
    font-size: 2.2rem;
    margin-bottom: 2rem;
    letter-spacing: 10px;
    padding-left: 10px;
  }
}

.highscore-body {
  background: transparent;
  cursor: pointer;
  position: relative;
}

.hs-row {
  font-size: 1.3rem;
 

  @media (min-width: $desktop-breakpoint) {
    font-size: 1.7rem;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    font-size: 1.3rem;
  }

  @media (min-height: 820px) {
    font-size: 1.8rem;
  }

  //iphone 11 pro & iphone X & iphone Xs
  @media only screen and (min-device-width: 375px) and (min-device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait) {
    font-size: 1.4rem;
  }
  //iphone 11 & iphone XR
  @media only screen and (device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) {
    font-size: 1.8rem;
  }
}

.hs-player {
  margin: 0;
  text-transform: uppercase;
  width: 17rem;

  color: $off-white;
  transition: 200ms color;

  @media (min-width: $desktop-breakpoint) {
    width: 20rem;
  }
}

.hs-player:hover {
  color: lighten($off-white, 20%);
}

.hs-score {
  text-align: right;
  margin: 0;
  color: $off-white;
}

.load-message {
  color: $orange;
  font-size: 1.5rem;
  margin-top: 10rem;
}
