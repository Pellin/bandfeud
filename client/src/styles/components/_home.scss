.new-game-button-container {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 10rem;
  margin-top: $desktop-top;

  @media (min-width: $desktop-breakpoint) {
    margin-top: 30rem;
  }
}

.new-game-button {
  background: $off-black;
  border: 2px solid $red;
  border-radius: 50px;
  box-shadow: 0 2px red, 0 2px inset black;
  color: $mustard;
  cursor: pointer;
  font-size: 1.9rem;
  outline: none;
  width: 9rem;
  height: 9rem;
  padding: 0;
  position: fixed;
  text-shadow: 0 1px black;

  @media (min-width: $desktop-breakpoint) {
    font-size: 2.6rem;
    width: 10rem;
    height: 10rem;
  }
}
