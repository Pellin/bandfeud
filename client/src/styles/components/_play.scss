.view-holder {
  background: $off-black;
  display: flex;
  justify-content: center;
  align-items: center;
  overflow: hidden;
}

.game-over-container {
  display: flex;
  flex-direction: column;
}

.game-over-message {
  color: $mustard;
  font-size: 10vw;
  margin: auto;
  position: absolute;
  text-align: center;
  top: 25rem;
  width: 100vw;
}

.game-container {
  align-items: space-between;
  background: $off-black;
  display: flex;
  justify-content: space-between;
  margin: 27rem 0 0 0;
  height: 35rem;
  width: 100%;
  position: fixed;

  @media (min-width: $desktop-breakpoint) {
    margin: 33rem 0 0 0;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    margin: 28rem 0 0 0;
  }

  @media (max-width: 568px) {
    margin: 25rem 0 0 0;
  }
  //iphone 11 pro & iphone X & iphone Xs
  @media only screen and (min-device-width: 375px) and (min-device-height: 812px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait) {
    margin: 27rem 0 20rem 0;
  }
  //iphone 11 & iphone XR
  @media only screen and (device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2) {
    margin: 27rem 0 20rem 0;
  }

  //iphone 11 pro max & iphone Xs Max
  @media only screen and (device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3) {
    height: 30rem;
    margin: 27rem 0 0 0;
  }
}

.game-container-firefox {
  background: $off-black;
  margin: 10rem 0 0 0;
  height: 35rem;
  width: 100%;
  position: relative;

  @media (min-width: $desktop-breakpoint) {
    margin: 15rem 0 0 0;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    margin: 12rem 0 0 0;
  }

  @media (max-width: 568px) {
    margin: 10rem 0 0 0;
  }
}

.band-list {
  height: 40rem;
}

.footer {
  background: $off-black;
  bottom: 2rem;
  position: fixed;
  height: 4rem;
  width: 100vw;
  border-top: 1px solid $off-white;
  padding-bottom: 1rem;
  z-index: 1000;

  @media (min-width: $desktop-breakpoint) {
    height: 7rem;
  }

  @media (max-height: 640px) and (min-width: 570px) {
    height: 5rem;
  }
}

.submit-band {
  background: $off-black;
  align-items: center;
  font-size: 2rem;
  justify-content: center;
  display: flex;
  flex-direction: column;
  width: 100vw;
  z-index: 1000;
}

.message-box {
  background: $off-black;
  font-size: 1.5rem;
  height: 1.5rem;
  margin: auto;
  bottom: 2rem;
  position: absolute;
  text-align: center;
  width: 100vw;

  @media (min-width: $desktop-breakpoint) {
    font-size: 2rem;
    height: 2rem;
  }
}
