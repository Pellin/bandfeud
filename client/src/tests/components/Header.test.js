import React from 'react';
import { shallow } from 'enzyme';

import { Header } from '../../components/Header';

it('should render Header correctly when inGame is true', () => {
  const wrapper = shallow(<Header inGame={true} used={['one', 'two']} />);

  expect(wrapper).toMatchSnapshot();
});

it('should render Header correctly when inGame is false', () => {
  const wrapper = shallow(<Header inGame={false} used={['one', 'two']} />);
  expect(wrapper).toMatchSnapshot();
});
