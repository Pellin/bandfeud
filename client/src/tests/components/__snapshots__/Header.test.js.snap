// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly when inGame is false 1`] = `
<div
  className="header-container"
>
  <div
    className="header-side"
  >
    <ForwardRef(MotionComponent)
      animate="show"
      className="i-link"
      initial={
        Object {
          "opacity": 0,
        }
      }
      transition={
        Object {
          "duration": 1.5,
        }
      }
      variants={
        Object {
          "hide": Object {
            "opacity": 0,
          },
          "show": Object {
            "opacity": 1,
          },
        }
      }
    >
      <Link
        to="/about/"
      >
        <ISvg />
      </Link>
    </ForwardRef(MotionComponent)>
  </div>
  <div
    className="header-center"
  />
  <div
    className="header-side"
  >
    <ForwardRef(MotionComponent)
      animate="show"
      className="crown-link"
      initial={
        Object {
          "opacity": 0,
        }
      }
      transition={
        Object {
          "duration": 1.5,
        }
      }
      variants={
        Object {
          "hide": Object {
            "opacity": 0,
          },
          "show": Object {
            "opacity": 1,
          },
        }
      }
    >
      <Link
        to="/highscores/"
      >
        <SvgCrown
          className="crown-right"
        />
      </Link>
    </ForwardRef(MotionComponent)>
  </div>
</div>
`;

exports[`should render Header correctly when inGame is true 1`] = `
<div
  className="header-container"
>
  <div
    className="header-side"
  />
  <div
    className="header-center"
  >
    <div
      className="header-logo"
    >
      <BFLogoSvg />
    </div>
  </div>
  <div
    className="header-side"
  />
</div>
`;
